src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Configuration/ConfigUpdaterSubStep.cs

```diff
@@ -90,7 +90,8 @@ protected override async Task<UpgradeStepInitializeResult> InitializeImplAsync(I
 
             try
             {
-                return await _configUpdater.IsApplicableAsync(context, _parentStep.ConfigFiles, token).ConfigureAwait(false)
+                return await IsApplicableAsync(context, token).ConfigureAwait(false)
```

Comment:
I'm not sure we need to check `IsApplicableAsync` here (despite the fact that we're calling _configUpdater.IsApplicableAsync - I've filed #427 to fix the fact that these two different concepts have the same name).

If a ConfigUpdaterStep isn't applicable, users shouldn't be calling Initialize at all. If they do, then this check here will just cause the step to show up as 'not yet finished' and needing to be applied, which is wrong. If we do want to check `IsApplicable` here, it probably should throw or return Failed if it comes back false.

Reply:
A substep (WebNamespaceConfigUpdater) that was hidden from the user was being asked if it was complete. Per our discussion I've implemented a filter in the `ConfigUpdaterStep.cs` that supports reset.